In [1]:
import os
for f in os.listdir('../fun/'): exec(open('../fun/'+f).read())
del f

# Load data.
load( '../out/d3-fight-level-transform.pkl' )
print( X.shape )

(4368, 165)


In [2]:
# train test split.
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    X, y, 
    random_state = 718,
    test_size = 0.25
)

In [3]:
# Fit decision tree.
from sklearn.ensemble import RandomForestClassifier
m = RandomForestClassifier( 
    n_estimators = 200, 
    min_samples_leaf = .1, 
    random_state = 841 
)
m.fit( X_train, y_train )

RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
                       max_depth=None, max_features='auto', max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=0.1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=200,
                       n_jobs=None, oob_score=False, random_state=841,
                       verbose=0, warm_start=False)

In [4]:
# accuracy against train data (in-model).
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( 
    y_train, 
    m.predict(X_train)
))

              precision    recall  f1-score   support

           0       0.00      0.00      0.00      1056
           1       0.68      1.00      0.81      2220

    accuracy                           0.68      3276
   macro avg       0.34      0.50      0.40      3276
weighted avg       0.46      0.68      0.55      3276



c:\users\bryce\documents\python37\lib\site-packages\sklearn\metrics\classification.py:1437: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


In [5]:
# accuracy against test data.
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( 
    y_test, 
    m.predict(X_test)
))

              precision    recall  f1-score   support

           0       0.00      0.00      0.00       384
           1       0.65      1.00      0.79       708

    accuracy                           0.65      1092
   macro avg       0.32      0.50      0.39      1092
weighted avg       0.42      0.65      0.51      1092



c:\users\bryce\documents\python37\lib\site-packages\sklearn\metrics\classification.py:1437: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


In [6]:
from sklearn.model_selection import cross_validate
kfold = cross_validate( 
    m, X, y, 
    cv = 10, 
    scoring = [ 'precision', 'recall' ], 
    n_jobs = -1 
)
print( 
  'Mean precision: %s \nMean recall %s' % 
  ( kfold['test_precision'].mean(), kfold['test_recall'].mean() ) 
)

Mean precision: 0.6703293934877081 
Mean recall 1.0


Test-set precision is 68% (predicting all-wins give 60%). Train/test is similar indicating no over-fitting. Move on to cross-validation: